spread.yaml

```
@@ -125,10 +126,55 @@ prepare: |
   cd "$SPREAD_PATH"
   snap install --classic concierge
 
+  if [[ $SPREAD_TASK == *"oauth"* ]]
+  then
+    # Running on IS-hosted runner; configure microk8s to use Docker Hub mirror
+    # Run before concierge prepare because of https://github.com/canonical/concierge/issues/75
+    sudo snap install microk8s  --classic
+    sudo snap install kubectl --classic 
+    sudo snap install yq
+    
+    if [[ -n "$DOCKERHUB_MIRROR" ]]
+    then
+      # Wait for microk8s to populate iptables
+      # https://chat.canonical.com/canonical/pl/jo5cg6wqjjrudqd5ybj6hhttee
+      until [[ -n $(iptables --list | grep -i "microk8s") ]]
+      do
+        echo "MicroK8s has not yet configured iptables."
+        sleep 10
+      done
+      tee /var/snap/microk8s/current/args/certs.d/docker.io/hosts.toml << EOF
+      server = "$DOCKERHUB_MIRROR"
+      [host."${DOCKERHUB_MIRROR#'https://'}"]
+      capabilities = ["pull", "resolve"]
+  EOF
+    fi
```

```suggestion
     tee /var/snap/microk8s/current/args/certs.d/docker.io/hosts.toml <<EOF
server = "$DOCKERHUB_MIRROR"
[host."${DOCKERHUB_MIRROR#'https://'}"]
capabilities = ["pull", "resolve"]
EOF
    fi
```
The lines in the EOF should not have any empty space in the beginning of line.

---

Fixed. Thank you.